test-lit/Semantics/common.sh

```diff
@@ -24,5 +24,16 @@ case $1 in
 esac
 USER_OPTIONS=`sed -n 's/^ *! *OPTIONS: *//p' $src`
 echo $USER_OPTIONS
-temp=`mktemp -d ./tmp.XXXXXX`
+# Set source directory depending on if it is a single file or multiple
+if [[ "$src" != "${src%[[:space:]]*}" ]]
+then
+  read -A $temp <<< $src
+  temp=$temp[1]
+  echo $temp
+  temp=$(echo "$temp" | cut -f 1 -d '.')
+else
+  temp=$(echo "$src" | cut -f 1 -d '.')
+fi
+temp="Outputs/$temp"
+mkdir -p $temp
```

Comment:
In the other PR, weren't you using `%t` and letting lit manage the temp directory? That seems simpler.

Reply:
I'm not 100% sure, but either lit doesn't actually do the thing I need it to do (cleaning up most likely), or judging from that if statement, it's more likely to do with the fact that this script doesn't know how to handle the input that lit now gives it due to manual changes in order to make the tests pass with lit.

Reply:
I can pass %t, but the shell scripts need to make the directory still within %t. Otherwise, they complain that the directory doesn't exist(which is true, because they're not looking at %t, something like %t/log instead), which lit can't handle as a result of this combination.

Reply:
%t is a unique _nonexistent_ path that tests can use for whatever they want. So you still need to create it, lit doesn't do that for you. In this case I do think you could use mkdir %t here.